src/components/thumbnail.test.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {Image, ViewPropTypes} from 'react-native';
+import PropTypes from 'prop-types';
+
+// Note: test renderer must be required after react-native.
+import renderer from 'react-test-renderer';
+
+import Thumbnail from './thumbnail';
+import {DEFAULT_WIDTH, TYPES} from './constants';
+import { getVideoId } from '../helpers/index';
+
+const testUrl = 'http://www.youtube.com/watch?v=abc123';
+
+describe('Thumbnail Component', () => {
+  test('snapshot matches', () => {
+    const tree = renderer.create(
+      <Thumbnail url={testUrl} />
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  test('initial state', () => {
+    const instance = renderer.create(
+      <Thumbnail url={testUrl} />
+    ).getInstance();
+
+    // TODO: Mock the result
```

Comment:
What do you mean by this?

Reply:
This test is testing the result of `getTestById` so this is a bit flaky if that function changes what it is doing. This test shouldn't care about the result specifically, rather that the contract was satisfied and it handled the result properly.